src/main/scala/piconot/language/parsing.scala

```diff
@@ -0,0 +1,214 @@
+package piconot.language
+
+import scala.language.postfixOps
+
+/**
+ * @author dhouck apinson
+ */
+
+/** Surroundings are declared by a capital char for the direction
+ *  followed by "open" or "closed" */
+sealed class Surroundings(val direction: Direction, val open: Boolean)
+case object Nopen extends Surroundings(North, true)
+case object Eopen extends Surroundings(East, true)
+case object Wopen extends Surroundings(West, true)
+case object Sopen extends Surroundings(South, true)
+case object Nclosed extends Surroundings(North, false)
+case object Eclosed extends Surroundings(East, false)
+case object Wclosed extends Surroundings(West, false)
+case object Sclosed extends Surroundings(South, false)
+case object Fopen extends Surroundings(Forward, true)
+case object Bopen extends Surroundings(Back, true)
+case object Lopen extends Surroundings(Left, true)
+case object Ropen extends Surroundings(Right, true)
+case object Fclosed extends Surroundings(Forward, false)
+case object Bclosed extends Surroundings(Back, false)
+case object Lclosed extends Surroundings(Left, false)
+case object Rclosed extends Surroundings(Right, false)
+
+/** Like a Surroundings, but for everything unspecified */
+object Anywhere
+
+/** The parser takes in an AST to be modified as the rules are parsed */
```

Comment:
I can't help but wonder if there's an easier way to implement this. I suppose you did a bit too though :)